cmd/metal-api/internal/service/network-service.go

```diff
@@ -439,6 +456,79 @@ func validateAdditionalAnnouncableCIDRs(additionalCidrs []string, privateSuper b
 	return nil
 }
 
+func validatePrefixesAndAddressFamilies(prefixes, destinationPrefixes []string, defaultChildPrefixLength metal.ChildPrefixLength, privateSuper bool) (metal.Prefixes, metal.Prefixes, metal.AddressFamilies, error) {
+
+	pfxs, addressFamilies, err := validatePrefixes(prefixes)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	// all DestinationPrefixes must be valid and from the same addressfamily
+	destPfxs, destinationAddressFamilies, err := validatePrefixes(destinationPrefixes)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if len(destinationAddressFamilies) > len(addressFamilies) {
+		return nil, nil, nil, fmt.Errorf("destination prefixes have more addressfamilies then prefixes")
+
+	}
+
+	if !privateSuper {
+		return pfxs, destPfxs, addressFamilies, nil
+	}
+
+	if len(defaultChildPrefixLength) == 0 {
+		return nil, nil, nil, fmt.Errorf("private super network must always contain a defaultchildprefixlength")
+	}
+
+	for af, length := range defaultChildPrefixLength {
+		fmt.Printf("af %s length:%d addressfamilies:%v", af, length, addressFamilies)
+		if !slices.Contains(addressFamilies, af) {
+			return nil, nil, nil, fmt.Errorf("private super network must always contain a defaultchildprefixlength per addressfamily:%s", af)
+		}
+
+		// check if childprefixlength is set and matches addressfamily
+		for _, p := range pfxs {
+			ipprefix, err := netip.ParsePrefix(p.String())
+			if err != nil {
+				return nil, nil, nil, fmt.Errorf("given prefix %v is not a valid ip with mask: %w", p, err)
+			}
+			if ipprefix.Addr().Is4() && af == metal.IPv6AddressFamily {
+				continue
+			}
+			if ipprefix.Addr().Is6() && af == metal.IPv4AddressFamily {
+				continue
+			}
+			if int(length) <= ipprefix.Bits() {
+				return nil, nil, nil, fmt.Errorf("given defaultchildprefixlength %d is not greater than prefix length of:%s", length, p.String())
+			}
+		}
+	}
+
+	return pfxs, destPfxs, addressFamilies, nil
+}
+
+func validatePrefixes(prefixes []string) (metal.Prefixes, metal.AddressFamilies, error) {
+	var (
+		result          metal.Prefixes
+		addressFamilies = make(map[metal.AddressFamily]bool)
+	)
+	for _, p := range prefixes {
+		prefix, ipprefix, err := metal.NewPrefixFromCIDR(p)
+		if err != nil {
+			return nil, nil, fmt.Errorf("given prefix %v is not a valid ip with mask: %w", p, err)
+		}
+		if ipprefix.Addr().Is4() {
+			addressFamilies[metal.IPv4AddressFamily] = true
+		}
+		if ipprefix.Addr().Is6() {
+			addressFamilies[metal.IPv6AddressFamily] = true
+		}
+		result = append(result, *prefix)
+	}
+	return result, lo.Keys(addressFamilies), nil
```

Comment:
you could also use `slices.Collect(maps.Keys(addressFamilies))`, with only the standard library functions without the need of "github.com/samber/lo"

Reply:
Ah, I see, lo is already used somewhere else, so it doesn't matter